File: FamilyTools.Web/src/app/models/template.ts
import { BaseModel } from './base-model';
import { AccountLine } from './account-line';

export class Template extends BaseModel {
  name: string;
  lines: AccountLine[];
  lifeTime: Date;

  constructor(name: string, lines: AccountLine[], lifeTime: Date, id?: number, creationDate?: Date, updateDate?: Date) {
    super(id, creationDate, updateDate);
    this.name = name;
    this.lines = lines;
    this.lifeTime = lifeTime;
  }
}

File: FamilyTools.EasyCompta/Model/Template.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FamilyTools.EasyCompta.Model
{
    [Table("Templates")]
    public class Template : BaseModel
    {
        public string Name { get; set; }

        public List<AccountLine> Lines { get; set; }

        public DateOnly LifeTime { get; set; }
    }
}
